interfaces/system_key_test.go

```diff
@@ -605,3 +635,146 @@ func (s *systemKeySuite) TestRemoveSystemKey(c *C) {
 	err = interfaces.RemoveSystemKey()
 	c.Assert(err, IsNil)
 }
+
+func (s *systemKeySuite) TestSystemKeyMismatchAdviceTrivial(c *C) {
+	mockedSkS := `
+{
+"build-id": "7a94e9736c091b3984bd63f5aebfc883c4d859e0",
+"apparmor-features": ["caps", "dbus", "more", "and", "more"]
+}
+`
+	s.AddCleanup(interfaces.MockSystemKey(mockedSkS))
+
+	c.Assert(interfaces.WriteSystemKey(interfaces.SystemKeyExtraData{}), IsNil)
+
+	mockedSk, err := interfaces.SystemKeyFromString(mockedSkS)
+	c.Assert(err, IsNil)
+
+	// compareing with self, so nothing to regenerate
+	act, err := interfaces.SystemKeyMismatchAdvice(mockedSk)
+	c.Assert(err, IsNil)
+	c.Check(act, Equals, interfaces.SystemKeyMismatchActionNone)
+}
+
+func (s *systemKeySuite) TestSystemKeyMismatchAdviceNFSHome(c *C) {
+	mockedSkS := `{
+"build-id": "7a94e9736c091b3984bd63f5aebfc883c4d859e0",
+"apparmor-features": ["caps", "dbus", "more", "and", "more"],
+"nfs-home": false
+}`
+	s.AddCleanup(interfaces.MockSystemKey(mockedSkS))
+
+	c.Assert(interfaces.WriteSystemKey(interfaces.SystemKeyExtraData{}), IsNil)
+
+	mockedSk, err := interfaces.SystemKeyFromString(`{
+"build-id": "7a94e9736c091b3984bd63f5aebfc883c4d859e0",
+"apparmor-features": ["caps", "dbus", "more", "and", "more"],
+"nfs-home": true
+}`)
+	c.Assert(err, IsNil)
+
+	// compareing with self, so nothing to regenerate
```

Comment:
```suggestion
	// comparing with self, so nothing to regenerate
```

Reply:
ah, this comment is completely wrong, updated now